### ISSUE TITLE
Feature request: Provide option to customise ErrorList HTML like ArrayFieldTemplate

### ISSUE THREAD

**Issue by elisechant** [mozilla-services/react-jsonschema-form#488]
### Description

It would be good to be able to write custom HTML for ErrorList at src/components/ErrorList.js in the same way we can for ArrayFieldTemplate.

**Comment by n1k0** [comment]
That would be a great feature to have indeed. Anybody willing to work on that?

**Comment by revolunet** [comment]
I also had to workaround that by creating a FieldTemplate with a custom ErrorList

would a `ErrorList` component props to the form do the trick ?

**Comment by n1k0** [comment]
I think we're talking about the general ErrorList, the one rendered at the top of the form, not the contextual field errors one, for which a field template works nicely. But yeah, having the ability to customize the top one would be good, and passing a custom ErrorList component as a form prop would probably work best.

**Comment by elisechant** [comment]
Here's how I'm working around this by hooking in to Form.renderErrors():

```
import React, {PureComponent, PropTypes} from 'react';
import Form from 'react-jsonschema-form';

class BaseForm extends Form {
  renderErrors() {
    if (this.props.onInternalStateChange) {
      const passedInProps = this.props.onInternalStateChange();
      const {globalErrors} = passedInProps;
      if (globalErrors) {
        // emulate what usually happens inside onSubmit
        this.setState({status: "initial", errors: [], errorSchema: {}});
        return <div className="panel panel-danger errors">
          <ul className="list-group">{
            globalErrors.map((error, i) => {
              return (
                <li key={i} className="list-group-item text-danger">{
                  error
                }</li>
              );
            })
          }</ul>
        </div>
      }
    }
    super.renderErrors();
  }
}


class SliceForm extends PureComponent {

  onSubmit({formData}) {
    const self = this;
    const payload = this.normalizePayload(formData);
    return this.props.handleSubmit(payload).then((response) => {
      // do stuff on success 
    }).catch(error => {
      self.setState({
        globalErrors: [error && error.message ? error.message : 'Submit failed']
      });
      // trigger an internal state change
      self.refs['form'].setState({'foo':'bar'});
    });
  }

  // return props that can be accessed inside internal state
  onInternalStateChange() {
    const {globalErrors} = this.state;
    return {
      globalErrors: globalErrors && globalErrors.length ? globalErrors : null
    }
  }

  render() {
    const {formData, schema, uiSchema} = this.state;

    return <BaseForm ref="form" noHtml5Validate={true}
              autocomplete="off"
              formData={formData}
              schema={schema}
              uiSchema={uiSchema}
              onInternalStateChange={this.onInternalStateChange.bind(this)} />;
  }
}
```

**Comment by n1k0** [comment]
Yeah that's too complicated, we should provide a simpler way. Who's willing to work on a PR? :)

**Comment by revolunet** [comment]
unless @elisechant wants to give a shot, i can try the form prop solution

**Comment by elisechant** [comment]
maybe a better workaround, you could also do it outside of jsonschema-form and use a callback ref to get access to jsonschems-form state like` <Form ref={(el) => this.form = el}`:

```
class MyForm extends PureComponent {

  onSubmit(formState) {
    const {formData} = formState;
    const self = this;
    const payload = this.normalizePayload(formData);

    return this.props.handleSubmit(payload).then((response) => {
      self.props.onSaveSuccess(response);
    }).catch(error => {
      self.setState({
        globalErrors: [error && error.message ? error.message : 'Submit failed']
      });
      // can access internal form state from ref
      self.form.setState({status: "initial", errors: [], errorSchema: {}});
      return false;
    });
  }

  renderGlobalErrors() {
    const {globalErrors} = this.state;
    if (globalErrors && globalErrors.length) {
      return <div className="panel panel-danger errors">
        <ul className="list-group">{
          globalErrors.map((error, i) => {
            return (
              <li key={i} className="list-group-item text-danger">{
                error
              }</li>
            );
          })
        }</ul>
      </div>
    }
    return null;
  }

  render() {
    const {formData, schema, uiSchema} = this.state;

    return (
      <div>
        {this.renderGlobalErrors()}
        <Form ref={(el) => this.form = el} noHtml5Validate={true}
              autocomplete="off"
              formData={formData}
              schema={schema}
              uiSchema={uiSchema}
              onSubmit={this.onSubmit.bind(this)}/>
      </div>
    )
  }
}
```

**Comment by n1k0** [comment]
Released in [v0.44.0](https://github.com/mozilla-services/react-jsonschema-form/releases/tag/v0.44.0).
